CRM/Report/Form/Member/Detail.php

```diff
@@ -331,7 +331,11 @@ public function from() {
              LEFT JOIN civicrm_membership_payment cmp
                  ON {$this->_aliases['civicrm_membership']}.id = cmp.membership_id
              LEFT JOIN civicrm_contribution {$this->_aliases['civicrm_contribution']}
-                 ON cmp.contribution_id={$this->_aliases['civicrm_contribution']}.id\n";
+                 ON cmp.contribution_id={$this->_aliases['civicrm_contribution']}.id
+             JOIN (SELECT MAX(id) id, contact_id  
+                FROM civicrm_contribution  WHERE civicrm_contribution.financial_type_id = 2
```

Comment:
@sqweets I think we shouldn't really be hard coding the financial_type_id here, can we not get that from the form values or is there no equivalent one? The other option is you could load the financial type from the membeship_type configuration